UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Filters/BespokeExchangeSetAuthorizationFilterAttribute.cs

```diff
@@ -45,10 +59,31 @@ public override async Task OnActionExecutionAsync(ActionExecutingContext context
             //If request is Bespoke exchange set and user is Non UKHO
             if (string.Equals(exchangeSetStandard, Common.Models.Enums.ExchangeSetStandard.s57.ToString(), StringComparison.OrdinalIgnoreCase))
             {
-                if (azureAdClientId != tokenAudience)
+                if (azureAdTenantId == tokenTenantId)
+                {
+                    if (azureAdClientId != tokenAudience)
+                    {
+                        context.HttpContext.Response.StatusCode = StatusCodes.Status403Forbidden;
+                        return;
+                    }
+                }
+
+                AzureAdB2C azureAdB2C = new AzureAdB2C
                 {
-                    context.HttpContext.Response.StatusCode = StatusCodes.Status403Forbidden;
-                    return;
+                    AudToken = tokenAudience,
+                    IssToken = tokenIssuer,
+                };
+
+                if (azureAdB2CHelper.IsAzureB2CUser(azureAdB2C, correlationId))
+                {
+                    var adminDomains = !string.IsNullOrEmpty(this.configuration["AdminDomains"]) ? new(this.configuration["AdminDomains"].Split(',').Select(s => s.Trim())) : new List<string>();
+                    var userDomain = context.HttpContext.User.FindFirstValue(ClaimTypes.Email)?.Split('@')[1].ToString();
+
+                    if (!adminDomains.Contains(userDomain))
```

Comment:
Please update check with case insensitive also do change this to ends with.

Reply:
Resolved as suggested.